src/test/java/keycontacts/logic/parser/ParserUtilTest.java

```diff
@@ -163,4 +163,19 @@ public void parseTime_validValueWithWhitespace_returnsTrimmedTime() throws Excep
         Time expectedTime = new Time(VALID_TIME);
         assertEquals(expectedTime, ParserUtil.parseTime(timeWithWhitespace));
     }
+    @Test
+    public void parseDay_validDay_success() throws ParseException {
+        assertEquals(new Day("Sunday"), ParserUtil.parseDay("Sunday"));
+    }
+
+    @Test
+    public void parseDay_validDayWithSpaces_success() throws ParseException {
+        assertEquals(new Day("Sunday"), ParserUtil.parseDay("    Sunday    "));
+    }
+
+    @Test
+    public void parseDay_invalidDay_failure() throws ParseException {
+        assertThrows(ParseException.class, Day.MESSAGE_CONSTRAINTS, () -> ParserUtil.parseDay("hello"));
+    }
+
```

Comment:
parseDay tests should already exist? Perhaps you meant to test parseDate? In that case, do follow the level of abstraction in the other test methods. For example, 
```suggestion
@Test
public void parseDate_validDate_success() throws ParseException {
    assertEquals(new Date(VALID_DATE), ParserUtil.parseDate(VALID_DATE));
}

@Test
public void parseDate_validDateWithSpaces_success() throws ParseException {
    assertEquals(new Date(VALID_DATE), ParserUtil.parseDate(WHITESPACE + VALID_DATE + WHITESPACE));
}

public void parseDate_invalidDate_failure() throws ParseException {
    assertThrows(ParseException.class, Date.MESSAGE_CONSTRAINTS, () -> ParserUtil.parseDate(INVALID_DATE));
}
```